p2p/src/main/java/bisq/network/p2p/storage/messages/AddOncePayload.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * This file is part of Bisq.
+ *
+ * Bisq is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * Bisq is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.network.p2p.storage.messages;
+
+/**
+ * Marker interface for messages which must not be added again after a remove message has been received (e.g. MailboxMessages).
+ */
+public interface AddOncePayload {
```

Comment:
Is there an issue number that this change fixes? After reviewing and refactoring this code I'm having trouble understanding the bug that would have needed this fix. I'd like to try and reproduce it in my tests, if possible.